### PR TITLE
Add/fix tests

### PR DESCRIPTION
**Pull request by nvie**
#### What does this PR do?

Adds tests for all endpoints, making sure we're preparing our requests in the correct fashion. I've made some minor changes to the client along the way (renaming arguments, making some optional, etc.)

#### How should this be manually tested?

Simply run `tox`. It should not make any actual HTTP requests, so the API service does not have to be running for this to work.

### CODE REVIEW

**Comment by nvie** [comment]
Almost there — more tomorrow!

screen shot 2015-03-20 at 01 28 13

**Comment by jonathanchu** [comment]
Ball so hard.  Writing tests at 02:30am. 

eslltye_jpg

File: yesgraph.py [hunk 1]
```diff
@@ -1,43 +1,83 @@
...
-        s.headers.update({
+    def _prepare_request(self, method, endpoint, data=None):
+        """Builds and prepares the complete request, but does not send it."""
+        headers = {
             'Authorization': 'Bearer {0}'.format(self.secret_key),
-            'Content-Type': 'application/json',
-        })
-        self.session = s
+        }
+        if method != 'GET':
```
Comment (ghickman) [inline yesgraph.py]: Seems like we could ignore this check and set the content-type on GETs too? Purely for simplicity.

Comment (nvie) [inline yesgraph.py]: I read the spec yesterday, but read it wrong. I thought that `Content-Type` should not be set on `GET` requests, but apparently, this is not the case. I'll remove this. Thanks.

Comment (nvie) [inline yesgraph.py]: Fixed in c1e9c7ba4109828f5cd263e953fbd214083250af.

**Comment by ghickman** [comment]
Merging because the travis breakage is fixed in master and the PR test run shows this.

**Pull request merged**